woocommerce/woocommerce-gutenberg-products-block
# Product Title block: add support global style

This PR adds support for global style for the `Product Title` block.

Now, the user can edit the style for:
- line-height
- font family
- font-weight
- text transform
- background-color
- margin
- font-size


Fixes #4965 

### Screenshots

![image](https://user-images.githubusercontent.com/4463174/148072903-c0553e9b-95a5-489f-9a62-f8044488d840.png)
*without global style*


![image](https://user-images.githubusercontent.com/4463174/148072776-1e9b69e2-b074-4c57-8643-8c64af3746b5.png)
*with global style*


### Testing

### Manual Testing

Check out this branch:

1. Install and enable the `Gutenberg` plugin on WordPress 5.9 site.
2. Install and enable the `Twenty Twenty-Two` theme.
3. Add the `Single Product block` (this block contains `Product Title Block`) to a post.
4. Get the focus on `Product Title block`.
5. On the right sidebar, personalize the styles of the block.
6. Go on the page and check if there are changes.
7. Reset to default using the `Reset` button from the different sections. 
8. Go to Dashboard and select Appearance > Editor (beta).
9. On the Editor page click on the `Styles` icon on the right-top corner.
10. Verify that the `Product Title block` is shown under the `Blocks` section. Personalize again the block.
11. Save your changes.
12. Go on the page created earlier and check if all styles are applied correctly.
13. Edit your previous post/page again.
14. Change again the styles.
15. Save your changes.
16. Check if these styles have priority over the styles from the Site Editor.

### Changelog

> Add support for the global style for the Product Title block.
